packages/cli/src/build/__tests__/__snapshots__/basic.ts.snap

```diff
@@ -33,7 +33,25 @@ exports.default = index;
 "
 `;
 
+exports[`basic: valid-package.esm.dev.js 1`] = `
+"var index = \\"something\\";
+
+export default index;
+"
+`;
+
 exports[`basic: valid-package.esm.js 1`] = `
+"'use strict';
+
+if (process.env.NODE_ENV === \\"production\\") {
+  module.exports = require(\\"./valid-package.esm.prod.js\\");
+} else {
+  module.exports = require(\\"./valid-package.esm.dev.js\\");
+}
+"
```

Comment:
I'm pretty sure that `.esm.js` file should not use `require` in it ;p so it sounds like a bug somewhere

Reply:
There is not way to do conditional imports in ESM. What would you expect here? Should I just remove this file or always use either prod or dev?

Reply:
A third option would be to import *BOTH* packages and then conditionally export the correct one. Hopefully the DCE feature of most bundlers will remove the extra code? Something like: 

```js
import Mod_prod from "./valid-package.esm.prod.js";
import Mod_dev from "./valid-package.esm.dev.js";

export default process.env.NODE_ENV === "production" ? Mod_prod : Mod_dev;
```

Reply:
As discussed somewhere else in this thread - let's just use leave `process.env.NODE_ENV` as is in this file. So we won't have dev & prod bundles for the ESM file (right now). This is not something I'm super happy about but it's the first step to move in the right direction in the future. Right now let's assume that ESM files are just targeting bundlers so it's their job to replace `process.env.NODE_ENV` with the correct value. I'd like to change that in the future but we need to do this carefully.

Reply:
I've realized that I forgot to respond directly to the last proposal here:

> A third option would be to import BOTH packages and then conditionally export the correct one. Hopefully, the DCE feature of most bundlers will remove the extra code?

Unfortunately, bundlers are pretty terrible as this (it ain't that easy to do it while maintaining safety guarantees) and we'd also risk running initialization code in those modules twice where the module author's intention is clearly for this kind of a code to be executed once.

Reply:
This file for esm is now just an alias for the dev build like so:

```js
export * from "./valid-package.esm.dev.js";
```